third_party/ibis/ibis_postgres/client.py

```diff
@@ -72,22 +72,23 @@ def connect(dbapi_connection, connection_record):
 
 
 def _metadata(self, query: str) -> sch.Schema:
-    raw_name = util.guid()
-    name = self._quote(raw_name)
-    type_info_sql = """\
-    SELECT
-    attname,
-    format_type(atttypid, atttypmod) AS type
-    FROM pg_attribute
-    WHERE attrelid = CAST(:raw_name AS regclass)
-    AND attnum > 0
-    AND NOT attisdropped
-    ORDER BY attnum"""
     with self.begin() as con:
-        con.exec_driver_sql(f"CREATE TEMPORARY VIEW {name} AS {query}")
-        type_info = con.execute(sa.text(type_info_sql).bindparams(raw_name=raw_name))
-        yield from ((col, _get_type(typestr)) for col, typestr in type_info)
-        con.exec_driver_sql(f"DROP VIEW IF EXISTS {name}")
+        cur = con.exec_driver_sql(f"select * from ({query}) t0 limit 0")
+        qry_cols = [
+            f"('{column.name}'::text, {column.type_code},"
+            + f"{column.table_oid if column.table_oid else 'NULL'}::int,"
+            + f"{column.table_column if column.table_column else 'NULL'}::int, {idx})"
+            for idx, column in enumerate(cur.cursor.description)
+        ]
+        type_info = con.exec_driver_sql(
+            f"""select name, CASE WHEN t0.attrelid is NULL
```

Comment:
A bit of a nitpick but I personally don't like to see a mixture of SQL keywords in upper and lower case. If it were me I'd either upper case tokens like "SELECT", "FROM UNNEST", etc or lower case the "CASE", "THEN", "ELSE" tokens. Just to be consistent.